test/test_helper.rb

```diff
@@ -2,63 +2,78 @@
 require 'kubernetes-deploy'
 require 'kubeclient'
 require 'pry'
-
 require 'minitest/autorun'
+require 'helpers/kubeclient_helper'
 
 ENV["KUBECONFIG"] ||= "#{Dir.home}/.kube/config"
 
-module KubeclientHelpers
-  def kubeclient
-    @kubeclient ||= begin
-      config = Kubeclient::Config.read(ENV["KUBECONFIG"])
-      unless config.contexts.include?("minikube")
-        puts "`minikube` context should be configured in your KUBECONFIG (#{ENV["KUBECONFIG"]})"
-      end
+module KubernetesDeploy
+  class TestCase < ::Minitest::Test
+    def setup
+      @logger_stream = StringIO.new
+      @logger = Logger.new(@logger_stream)
+      KubernetesDeploy.logger = @logger
+    end
 
-      client = Kubeclient::Client.new(
-        config.context.api_endpoint,
-        config.context.api_version,
-        {
-          ssl_options: config.context.ssl_options,
-          auth_options: config.context.auth_options
-        }
-      )
-      client.discover
-      client
+    def teardown
+      @logger_stream.close
     end
   end
-end
 
-Minitest::Test.include(KubeclientHelpers)
+  class IntegrationTest < KubernetesDeploy::TestCase
+    include KubeclientHelper
+
+    def run
+      @namespace = TestProvisioner.claim_namespace(self.name)
+      super
+    ensure
+      TestProvisioner.delete_namespace(@namespace)
+    end
```

Comment:
Wow, nice hacking.. 👍  I bet this is going to save someone a lot of time at some point.